### PR TITLE
Add StatusCode helpers to ControllerBase

### LINKED ISSUE #4149: Add Controller helper methods for StatusCode(int)/StatusCode(int,object)
I feel like I've answered questions about `ObjectResult` + status code a million times by now. We should make this more visible.

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Add some helper methods for returning StatusCode results and Object results with status codes.

Fixes #4149.

### CODE REVIEW

**Comment by ryanbrandenburg** [comment]
CC @rynowak

**Comment by rynowak** [comment]
These new methods should have unit tests since they are on `ControllerBase`

**Comment by ryanbrandenburg** [comment]
@pranavkm would also like doc comments.

**Comment by ryanbrandenburg** [comment]
:up::date:

File: test/Microsoft.AspNetCore.Mvc.Core.Test/ControllerBaseTest.cs [hunk 1]
```diff
@@ -1061,6 +1061,59 @@ public void Controller_Content_CharsetInContentType_IsUsedForEncoding()
...
+        [InlineData(200, true)]
+        [InlineData(0, false)]
+        [InlineData(-200, false)]
+        public void Controller_StatusCode_SetsStatusCode(int statusCode, bool shouldSucceed)
+        {
+            // Arrange
+            var statusCodeController = new StatusCodeController();
+
+            // Act
+            var result = (StatusCodeResult)statusCodeController.StatusCode_Int(statusCode);
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/ControllerBaseTest.cs]: You can get rid of these casts by just returning the concrete type.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/ControllerBaseTest.cs [hunk 2]
```diff
@@ -1061,6 +1061,59 @@ public void Controller_Content_CharsetInContentType_IsUsedForEncoding()
...
+        }
+
+
+
+        [Theory]
+        [InlineData(497, true)]
+        [InlineData(200, true)]
+        [InlineData(0, false)]
+        [InlineData(-200, false)]
+        public void Controller_StatusCode_SetsStatusCode(int statusCode, bool shouldSucceed)
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/ControllerBaseTest.cs]: What is this test for?

File: src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs [hunk 3]
```diff
@@ -204,6 +204,32 @@ public ClaimsPrincipal User
...
+        public virtual StatusCodeResult StatusCode(int statusCode)
+        {
+            return new StatusCodeResult(statusCode);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="ObjectResult"/> object by specifying a <paramref name="statusCode"/> and <paramref name="value"/>
+        /// </summary>
+        /// <param name="statusCode">The status code to set on the response.</param>
+        /// <param name="value">The value to set on the <see cref="ObjectResult"/> result.</param>
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs]: Remove the word `result` at the end?

**Comment by ryanbrandenburg** [comment]
:up::date:

File: test/Microsoft.AspNetCore.Mvc.Core.Test/ControllerBaseTest.cs [hunk 4]
```diff
@@ -1061,6 +1061,56 @@ public void Controller_Content_CharsetInContentType_IsUsedForEncoding()
...
+        public void Controller_StatusCode_SetsStatusCode(int statusCode)
+        {
+            // Arrange
+            var statusCodeController = new StatusCodeController();
+
+            // Act
+            var result = statusCodeController.StatusCode_Int(statusCode);
+
+            // Assert
+
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/ControllerBaseTest.cs]: Extra line

**Comment by rynowak** [comment]
:shipit:

**Comment by ajaybhargavb** [comment]
:shipit:

**Pull request merged**